src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2059,8 +2059,6 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnDnsResolveResult(otE
         return;
     }
 
-    ThreadStackMgrImpl().LockThreadStack();
```

Comment:
Why this was lock here at the first place ?

Reply:
I am not sure why it was there in the first place. But this function is called as a callback when the resolve is completed from `ThreadTaskMain` task. The task  only runs `ProcessThreadActivity` and is already protected by the lock therefore creating the deadlock.

Reply:
Should it also be removed from `OnDnsBrowseResult` then?

Reply:
I missed that. I assume yes. Let me test that

Reply:
Removed the lock for OnDnsBrowseResult also. Validated that it works with `dns browse commissionable` which also made me catch a task overflow in the EFR32. fixed that also.